docs/probing.md

```diff
@@ -0,0 +1,318 @@
+let network_graph = Arc::new(disk::read_network(Path::new(&network_graph_path), args.network, logger.clone()));
+
+// Initialize the Scorer.
+let scorer_path = format!("{}/scorer", ldk_data_dir.clone());
+let scorer = Arc::new(RwLock::new(disk::read_scorer(
+    Path::new(&scorer_path),
+    Arc::clone(&network_graph),
+    Arc::clone(&logger),
+)));
+```
+
+To create the ```scorer```, the above code utilizes a ```read_scorer``` function, which resides in our application's ```disk.rs``` file. The function has been provided below. In it, you can see that we are first attempting to read the scorer from our application's data folder. If a scorer is available, we will use that. If not, we'll create a new scorer. It's worth noting that, if we decided to retrieve a scorer via an alternative method such as an HTTP server, we could adjust this function to include that logic.
+
+```rust
+pub(crate) fn read_scorer(
+	path: &Path, graph: Arc<NetworkGraph>, logger: Arc<FilesystemLogger>,
+) -> ProbabilisticScorer<Arc<NetworkGraph>, Arc<FilesystemLogger>> {
+	let params = ProbabilisticScoringDecayParameters::default();
+	if let Ok(file) = File::open(path) {
+		let args = (params.clone(), Arc::clone(&graph), Arc::clone(&logger));
+		if let Ok(scorer) = ProbabilisticScorer::read(&mut BufReader::new(file), args) {
+			return scorer;
+		}
+	}
+	ProbabilisticScorer::new(params, graph, logger)
+}
+```
+
+#### Setting Up Our Probing Logic
+For this example, we'll build a simple prober that fetches a random node from our ```NetworkGraph``` and attempts to send that node a random payment - up to 500,000,000 milli-satoshis. We'll start by creating a function that fetches a random node from our ```NetworkGraph```. If we cannot find a node, the function will return, and we will not probe any nodes. If we're able to identify a node to probe, then we'll select a random payment amount of up to 500,000,000 milli-satoshis. As a reminder, 1 satoshi = 1,000 milli-satoshis. Finally, we'll get the public key for the node we selected and call ```send_probe```, which is defined below.
+
+```rust
+fn send_rand_probe(
+    channel_manager: &ChannelManager, graph: &NetworkGraph,
+    logger: &disk::FilesystemLogger,
+    scorer: &RwLock<ProbabilisticScorer<Arc<NetworkGraph>, Arc<disk::FilesystemLogger>>>
+) {
+    let recipient = {
+	// Get read-only lock on graph.
+        let read_only_graph = graph.read_only();
+
+	// Check if graph has any nodes we can probe. If none, return.
+        if read_only_graph.nodes().is_empty() { 
+            return; 
+        }
+        // Select random node from the graph.
+        let mut it = read_only_graph.nodes().unordered_iter().skip(::rand::random::<usize>() % read_only_graph.nodes().len());
+        it.next().unwrap().0.clone()
+    };
+    
+    // Generate a random amount for the probe - up to 500,000,000 milli-satoshis.
+    let amt = ::rand::random::<u64>() % 500_000_000;
```

Comment:
It might be worth noting somewhere how a more advance prober might choose an amount. For instance, it could try to hone in on the exact liquidity balance by choosing an amount that falls between the min and max liquidity estimate of a channel. Probing below / above the the min / max wouldn't yield any useful information. Although, changing the amount could result in a different path being chosen. To chose an exact path, they could use `lightning::routing::router::build_route_from_hops`. https://docs.rs/lightning/latest/lightning/routing/router/fn.build_route_from_hops.html